File: taskManagerController/src/main/java/com/example/taskManagerController/service/TaskService.java
package com.example.taskManagerController.service;
import com.example.taskManagerController.model.Task;
import jakarta.annotation.Nonnull;
import org.springframework.lang.NonNull;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.time.LocalDate;

@Service
public class TaskService {

    private final List<Task> tasks = new ArrayList<>();

    public TaskService() {
        tasks.add(new Task(1L, "Hiking", LocalDate.of(2025, 6, 5), "Okutama", true));
        tasks.add(new Task(2L, "Camping", LocalDate.of(2025, 6, 6), "Yamanashi", true));
        tasks.add(new Task(3L, "Snowboarding", LocalDate.of(2025, 6, 7), "Nozawa", false));
        tasks.add(new Task(4L, "Work", LocalDate.of(2025, 6, 8), "Day and PT", false));
        tasks.add(new Task(5L, "Cook", LocalDate.of(2025, 6, 9), "Breakfast, Lunch, Dinner", false));
    }

    public List<Task> getAllTasks() {
        return tasks;
    }

    public Optional<Task> getTaskById(Long id) {
        if (id == null) {
            throw new IllegalArgumentException("Task ID for retrieval cannot be null.");
        }
        return tasks.stream().filter(task -> task.getId().equals(id)).findFirst();
    }

    public Task addTask(Task task) {
        if (task == null) {
            throw new IllegalArgumentException("Task to add cannot be null.");
        }
        task.setId(generateNextId());
        tasks.add(task);
        return task;
    }

    public Optional<Task> updateTask(Long id, Task updatedTask) {
        if (id == null) {
            throw new IllegalArgumentException("Task ID for update cannot be null.");
        }
        if (updatedTask == null) {
            throw new IllegalArgumentException("Updated task object cannot be null.");
        }

        return getTaskById(id).map(task -> {
            task.setTitle(updatedTask.getTitle());
            task.setDate(updatedTask.getDate());
            task.setDescription(updatedTask.getDescription());
            task.setIsCompleted(updatedTask.getIsCompleted());
            return task;
        });
    }

    public boolean deleteTask(Long id) {
        if (id == null) {
            throw new IllegalArgumentException("Task ID for deletion cannot be null.");
        }
        return tasks.removeIf(task -> task.getId().equals(id));
    }

    private Long generateNextId() {
        return tasks.stream().mapToLong(Task::getId).max().orElse(0L) + 1;
    }
}

File: src/main/kotlin/com/example/taskManagerController/service/TaskService.kt
package com.example.taskManagerController.service

import com.example.taskManagerController.model.Task
import com.example.taskManagerController.dto.TaskRequest
import com.example.taskManagerController.dto.TaskResponse
import org.springframework.stereotype.Service
import java.time.LocalDate
import java.util.UUID

@Service
class TaskService {

    private var tasks: MutableList<Task> = mutableListOf()

    init {
        resetInitialTasks()
    }

    fun resetInitialTasks() {
        tasks = mutableListOf(
        Task(UUID.fromString("c016e41b-7a32-4d2b-8a8b-1a2b3c4d5e6f"), "Hiking", LocalDate.of(2025, 6, 5), "Okutama", true),
        Task(UUID.fromString("e72f3a8c-9d1e-4f0a-b1c2-3d4e5f6a7b8c"), "Camping", LocalDate.of(2025, 6, 6), "Yamanashi", true),
        Task(UUID.fromString("f8d9b0c1-2e3f-4a5b-6c7d-8e9f0a1b2c3d"), "Snowboarding", LocalDate.of(2025, 6, 7), "Nozawa", false),
        Task(UUID.fromString("1a2b3c4d-5e6f-7a8b-9c0d-1e2f3a4b5c6d"), "Work", LocalDate.of(2025, 6, 8), "Day and PT", false),
        Task(UUID.fromString("2a3b4c5d-6e7f-8a9b-0c1d-2e3f4a5b6c7d"), "Cook", LocalDate.of(2025, 6, 9), "Breakfast, Lunch, Dinner", false)
        )
    }

    fun getAllTasks(): List<Task> {
        return tasks.toList()
    }

    fun getTaskById(id: UUID): Task? {
        return tasks.firstOrNull { it.id == id }
    }

    fun addTask(request: TaskRequest): TaskResponse {
        val newTaskId = UUID.randomUUID()

        val newTask = Task(
            id = newTaskId,
            title = request.title,
            date = request.date,
            description = request.description,
            isCompleted = request.isCompleted
        )

        require(getTaskById(newTask.id) == null) { "Generated Task ID ${newTask.id} already exists. Please try again." }

        tasks.add(newTask)

        return TaskResponse(
            id = newTask.id,
            title = newTask.title,
            date = newTask.date,
            description = newTask.description,
            isCompleted = newTask.isCompleted,
            userId = request.userId
        )
    }

    fun updateTask(id: UUID, updatedTask: Task): Task? {
        require(updatedTask.id == id) { "Task object ID (${updatedTask.id}) must match the provided ID ($id) for update." }

        return tasks.firstOrNull { it.id == id }?.let { existingTask ->
            existingTask.title = updatedTask.title
            existingTask.date = updatedTask.date
            existingTask.description = updatedTask.description
            existingTask.isCompleted = updatedTask.isCompleted
            existingTask
        }
    }

    fun deleteTask(id: UUID): Boolean {
        return tasks.removeIf { it.id == id }
    }
}
